addons/product/models/product_product.py

```diff
@@ -436,6 +434,224 @@ def _onchange_uom_id(self):
+
+        ########################################
+        # Search product.template or create it #
+        ########################################
+
+        product_templates = PT.search(
+            Domain('name', 'in', [vals['name'] for vals in with_import_values]) |
+            Domain('id', 'in', [vals.get('product_tmpl_id') for vals in with_import_values]))
+        id2template = dict(zip(product_templates.ids, product_templates))  # id => Record<'product.template'>
+        name2template = dict(zip(product_templates.mapped('name'), product_templates))  # template_name => Record<'product.template'>
+
+        template_vals_list = {}
+        for vals in with_import_values:
+            name = vals['name']
+            if name and name not in name2template and name not in template_vals_list:
+                template_vals_list[name] = {key: value for key, value in vals.items() if PT._fields[key].required}
+
+        # create the first product to have default values
+        created_templates = PT.with_context(create_product_product=True).create(list(template_vals_list.values()))
+        for rec in created_templates:
+            id2template[rec.id] = rec
+            name2template[rec.name] = rec
+
+        field_names = list(with_import_values[0])
+        default_values = {
+            values['product_tmpl_id'][0]: values
+            for values in (product_templates.product_variant_id + created_templates.product_variant_id).read(fields=['product_tmpl_id'] + field_names)
+        }
+
+        PP.search(Domain('product_tmpl_id', 'in', id2template.keys()) & Domain('product_template_attribute_value_ids', '=', False)).unlink()
+
+        ############################################################################
+        # Create product.template.attribute.line if not exists or update value_ids #
+        ############################################################################
+
+        pt_to_attribute_to_values = defaultdict(list)  # (template_id, attribute_id) => list(Record<'product.attribute.value'>)
+        for vals in with_import_values:
+            for value in vals['import_attribute_values'].split(','):
+                attribute_name, value_name = value.split(':', 1)
+                pt = id2template.get(vals.get('product_tmpl_id')) or name2template.get(vals['name'])
+                pav = pa_pav_records[attribute_name.strip(), value_name.strip()]
+                pt_to_attribute_to_values[pt.id, pav.attribute_id.id].append(pav)
+        domain = Domain(False)
+        for template_id, attribute_id in pt_to_attribute_to_values:
+            domain |= (Domain('product_tmpl_id', '=', template_id) &
+                        Domain('attribute_id', '=', attribute_id))
+        template_attribute_to_ptal = {  # (template_id, attribute_id) => ptal
+            (ptal.product_tmpl_id.id, ptal.attribute_id.id): ptal
+            for ptal in PTAL.search(domain)
+        }
```

Comment:
FYI, it is supported to have multiple attribute lines with the same attribute.
Dunno how it would behave in your import logic, probably worth throwing some kind of exceptions if unsupported in the import and someone tries to do it.